RestrictEvents/RestrictEvents.cpp

```diff
@@ -263,6 +260,54 @@ struct RestrictEventsPolicy {
 		return true;
 	}
 
+	static void getBlockedProcesses() {
+		// Updates procBlacklist with list of processes to block
+		char duip[128] { "auto" };
+		if (PE_parse_boot_argn("revblock", duip, sizeof(duip))) {
+			DBGLOG("rev", "read revblock from boot-args");
+		} else if (readNvramVariable(NVRAM_PREFIX(LILU_VENDOR_GUID, "revblock"), u"revblock", &EfiRuntimeServices::LiluVendorGuid, duip, sizeof(duip))) {
+			DBGLOG("rev", "read revblock from NVRAM");
+		}
+
+		char *value = reinterpret_cast<char *>(&duip[0]);
+		value[sizeof(duip) - 1] = '\0';
+		int i = 0;
+
+		// Disable notification prompts for mismatched memory configuration on MacPro7,1
+		if (strstr(value, "pcie", strlen("pcie")) || strstr(value, "auto", strlen("auto"))) {
+			if (getKernelVersion() >= KernelVersion::Catalina) {
+				DBGLOG("rev", "disabling PCIe memory notification");
+				procBlacklist[i] = (char *)"/System/Library/CoreServices/ExpansionSlotNotification";
+				procBlacklist[i+1] = (char *)"/System/Library/CoreServices/MemorySlotNotification";
+				i += 2;
+			}
+		}
+
+		// MacBookPro9,1 and MacBookPro10,1 GMUX fails to switch with 'displaypolicyd' active in Big Sur and newer
+		if (strstr(value, "gmux", strlen("gmux"))) {
```

Comment:
Is this (nearly) mandatory on those MBP models? If so, how about automatically turning it on via SMBIOS info?

Reply:
It's a hardware issue with the GMUX, someone on a hackintosh or spoofing from a different Mac shouldn't have the patch enabled